modules/core/src/lib/attribute-manager.js

```diff
@@ -422,6 +425,15 @@ export default class AttributeManager {
     return needsUpdate;
   }
 
+  _setGenericAttributes({data, props, context}) {
+    const {attributes} = this;
+
+    for (const attributeName in attributes) {
```

Comment:
Why not iterate over attributes provided, check their values and call the appropriate method rather than calling this on all values?

Reply:
This is just following the existing pattern (`_checkExternalBuffers`, `_checkIfBuffersNeedUpdating`, `_allocateBuffers`, `_updateBuffers` each iterate through all attributes). I can refactor in a follow up PR.